Source/Events.Store.MongoDB/Processing/Streams/UnsupportedStateDerivedFromAbstractStreamProcessorState.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Dolittle.Runtime.Events.Processing.Streams;
+
+namespace Dolittle.Runtime.Events.Store.MongoDB.Processing.Streams
+{
+    /// <summary>
+    /// Exception that gets thrown when trying to use an unsuported <see cref="IStreamProcessorState"/> and <see cref="IStreamProcessorId"/> and .
+    /// </summary>
+    public class UnsupportedStateDerivedFromAbstractStreamProcessorState : Exception
```

Comment:
Should this exception extend a more generic exception defined outside the MongoDB event store implementation?

Reply:
Removed this particular exception, some of the mongo exceptions extend the `EventStoreConsistencyError` now too